src/trace.rs

```diff
@@ -0,0 +1,212 @@
+// For completeness, wrappers around all of tracing's public macros are provided, even if they are
+// not used at the present time.
+#![allow(unused_macros)]
+
+//! Internal Tracing macro module
+//!
+//! The [`trace`][crate::trace] module is an internal module that contains wrapper macros encapsulating
+//! [`tracing`][tracing]'s macros. These macros allow for conditional expansion of
+//! [`tracing`][tracing] macros during compilation when the `tracing` feature is enabled, or trimming
+//! them when the feature is disabled, all in a concise manner.
+//!
+//! The macros from the [`trace`][crate::trace] module are declared by default and can be used throughout the
+//! crate. However, as the contents of these macros are conditionally compiled, they are effectively trimmed
+//! during inline expansion when the `tracing` feature is disabled.
+//!
+//! # Unstable
+//!
+//! The [`tracing`][tracing] module is currenty **unstable**, hence the existence of this module. As a
+//! result, hyper's [`tracing`][tracing] logs are only accessibe if `--cfg hyper_unstable_tracing` is
+//! passed to `rustc` when compiling. The easiest way to do that is through setting the `RUSTFLAGS`
+//! enviornment variable.
+//!
+//! # Building
+//!  
+//! Enabling [`trace`][crate::trace] logs, can be done with the following `cargo` command, as of
+//! version `1.64.0`:
+//!
+//! ```notrust
+//! RUSTFLAGS="--cfg hyper_unstable_tracing" cargo rustc --features client,http1,http2,tracing --crate-type cdylib
+//! ```
+
+macro_rules! debug {
+    ($($arg:tt)+) => {
+        #[cfg(feature = "tracing")]
+        {
+            #[cfg(not(hyper_unstable_tracing))]
+            compile_error!(
+                "\
+                The `tracing` feature is unstable, and requires the \
+                `RUSTFLAGS='--cfg hyper_unstable_tracing'` environment variable to be set.\
+            "
+            );
```

Comment:
this will result in the same compile error being emitted for *every* use of a tracing macro, if the unstable cfg is unset. this means there will be a giant pile of the same error in the rustc output. i wonder if it would be nicer to instead just stick the compile error in this module, so that we generate a single error if the feature flag is enabled without the unstable cfg?

Reply:
I agree, I will make the change.